lib/parseArgs.js

```diff
@@ -0,0 +1,91 @@
+const chalk = require('chalk');
+const commandLineArgs = require('command-line-args');
+const path = require('path');
+
+const optionDefinitions = [
+  {
+    name: 'scriptName',
+    defaultOption: true
+  },
+  {
+    name: 'env',
+    alias: 'e',
+    type: String,
+    defaultValue: 'production'
+  },
+  {
+    name: 'tenant',
+    alias: 't',
+    type: String,
+    defaultValue: ''
+  },
+  {
+    name: 'build',
+    alias: 'b',
+    type: String
+  },
+  {
+    name: 'config',
+    alias: 'c',
+    type: String,
+    defaultValue: 'sku.config.js'
+  }
+];
+
+/**
+ * Supports parsing args that look like:
+ * [node, scripts/foo.js, arg1, arg2]
+ * [node, sku, foo, arg1, arg2]
+ *
+ * @param {String[]} args - should look like process.argv
+ */
+module.exports = _args => {
+  const args = [..._args];
+
+  // The first argv is always the node executable, which we don't care about.
+  args.shift();
+
+  // If you're running `sku something` then the next arg is the sku executable, which we don't
+  // care about either.
+  if (args[0].match(/bin\/sku(\.js)?/)) {
+    args.shift();
+  }
+
+  // At this point, args should be either
+  // [script/foo.js, arg1, arg2] or
+  // [foo, arg1, arg2]
+  const options = commandLineArgs(optionDefinitions, {
+    argv: args,
+    stopAtFirstUnknown: true,
+    partial: true
+  });
+
+  let { scriptName, _unknown: argv = [] } = options;
+
+  // Detect if you're trying to execute a script directly...
+  const scriptNameRegex = /scripts[\/\\]([\w-]*)\.js$/i;
```

Comment:
Is this something we want to support? Why do we need this?

Reply:
Good question! I'm happy to enforce using the `sku` executable; my first go at this did exactly that. However it's the same amount of work to detect and adapt as it is to prevent, and I figured that's more flexible.

I don't have any evidence for users running scripts directly, the only example was directly was in our tests, which I've updated here too.

Reply:
IMHO I think we're better off restricting it, just to keep things more predictable.

Reply:
After a chat @markdalgleish and I decided to not support the `node scripts/lint.js` API.